### PR TITLE
lava-docs: create autolinks

### PR DESCRIPTION
**Pull request by jroimartin**
This PR creates autolinks in the imported Lava help topics. This feature is not supported by mdbook as explained in https://github.com/rust-lang/mdBook/issues/1114.

Before,

image

After,

image

### CODE REVIEW

File: autolinks.bash [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2024 Adevinta
+
+# substitute_autolinks creates autolinks in the provided file if the
+# autolink ends with the provided delimiter.
+substitute_autolinks() {
+	local file=$1
+	local delim=$2
+
+	sed -i -E 's/(^|^[^\t].*?[[:space:]])(https?:\/\/[[:alnum:][:punct:]]+?)('"${delim}"')/\1<\2>\3/gm' "${file}"
```
Comment (jesusfcr) [inline autolinks.bash]: This doesn't works on mac.

- gsed should be installed/used instead.
- Also the dependencies are installed for linux.

I understand this is not a requirement so no big deal.

File: testdata/autolinks.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+https://example.org/path
+https://example.org/path/
+
+Link with parentheses https://example.org/path_(foo) and suffix.
+Link with parentheses https://example.org/path_(foo)/ and suffix.
+
+Autolink <https://example.org/path>
+Autolink <https://example.org/path/>
+
+	Code https://example.org/path
```
Comment (jesusfcr) [inline testdata/autolinks.md]: The typical markdown as below will be translated.

```code
curl https://www.google.com
```

I guess this is just for lava and lava will never generate this kind of markdown.

File: build.bash [hunk 3]
```diff
@@ -53,6 +55,7 @@ cat pages.json | jq --compact-output '.[]' | while read -r row; do
 	mkdir -p "$(dirname "${file}")"
 	"${install_dir}/lava" help "${topic}" > "${file}"
 	fix_headers "${file}"
+	fix_autolinks "${file}"
```
Comment (jesusfcr) [inline build.bash]: Why don't you create a preprocessor instead?
https://rust-lang.github.io/mdBook/format/configuration/preprocessors.html

**Review by jesusfcr — APPROVED** [review]
LGTM, just some comments I guess you are already aware of.

Apart from that the docs publication could be automated by generating the pages.json from `lava help` and using dependabot to trigger the execution every time a new release is published.

File: testdata/autolinks.md (hunk 2 above)
Comment (jroimartin) [reply testdata/autolinks.md]: Exactly. That format is not used in the Lava documentation.

File: build.bash (hunk 3 above)
Comment (jroimartin) [reply build.bash]: I just followed the current approach. We can give a look to preprocessors in future itrations.

File: autolinks.bash (hunk 1 above)
Comment (jroimartin) [reply autolinks.bash]: 1060083c9fd5e6c66398047ca3c3c42ccb5e2b8a moves build and test to docker. PTAL.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2024 Adevinta
+
+version: "2.4"
```
Comment (jesusfcr) [inline docker-compose.yml]: Remove obsolete `version` - https://github.com/compose-spec/compose-spec/blob/main/spec.md#version-and-name-top-level-elements

**Review by jesusfcr — CHANGES_REQUESTED** [review]
Just a comment

File: docker-compose.yml (hunk 4 above)
Comment (seilagamo) [reply docker-compose.yml]: I think it is obsolete if the build detects that can execute `docker compose`, but is mandatory if it has to execute `docker-compose`

**Review by jesusfcr — APPROVED** [review]
LGTM

File: docker-compose.yml (hunk 4 above)
Comment (jesusfcr) [reply docker-compose.yml]: Ok then, not a big deal.

File: docker/build.bash [hunk 5]
```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bash
+# Copyright 2024 Adevinta
+
+set -e -u
+
+cd "$(dirname $0)/.."
+
+MDBOOK_VERSION=${MDBOOK_VERSION:-v0.4.40}
```
Comment (seilagamo) [inline docker/build.bash]: I think this variable could be set in the same way as LAVA_VERSION

**Review by seilagamo — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 4 above)
Comment (jroimartin) [reply docker-compose.yml]: Removed. I tested it with docker-compose v1.29.2, which is the version packaged by Fedora Linux 40, and it works just fine.

File: docker/build.bash (hunk 5 above)
Comment (jroimartin) [reply docker/build.bash]: Fixed.

**Review by seilagamo — APPROVED** [review]
LGTM

**Pull request merged**